src/server/sendLayoutHTTPResponse.js

```diff
@@ -291,12 +348,25 @@ function serializeFragment({ node, bodyStream, renderOptions }) {
     throw Error(`<fragment> has unknown name`);
   }
 
-  bodyStream.add(
-    renderResultToStream(
-      () => renderOptions.renderFragment(attr.value),
+  let fragmentStream;
+  try {
+    fragmentStream = valueToStream(
+      renderOptions.renderFragment(attr.value),
       `Fragment ${attr.value}`
-    )
-  );
+    );
+  } catch (err) {
+    fragmentStream = renderError(`Fragment ${attr.name}`, err);
+  }
+
+  bodyStream.add(fragmentStream);
+}
+
+/**
+ *
+ * @param {SerializeArgs} serializeArgs
+ */
+function serializeAssets({ assetsStream, bodyStream }) {
+  bodyStream.add(assetsStream);
```

Comment:
I think this will help in ensuring that the assets are serialized in the correct order in the HTML.

```suggestion
  bodyStream.add(assetsStream);
  assetsStream.end();
```

Reply:
Did try this initially without much success and later relaized the issue was related to the CustomParser.js